ext/pgsql/pgsql.c

```diff
@@ -4250,7 +4268,14 @@ PHP_PGSQL_API zend_result php_pgsql_meta_data(PGconn *pg_link, const zend_string
 		name = PQgetvalue(pg_result,i,0);
 		add_assoc_zval(meta, name, &elem);
 	}
+
+	add_assoc_bool_ex(meta, pg_cache_field, sizeof(pg_cache_field) - 1, false);
+
+	tmp = (zval *)safe_emalloc(1, sizeof(zval), 0);
+	ZVAL_COPY(tmp, meta);
+	zend_hash_add(zmeta, table_name, tmp);
```

Comment:
This leaks because `meta` here is still `emalloc`ed. You'd need to copy the entire array to "process-space" recursively. May I suggest making the cache request-bound? This problem goes away, along with potential cache-invalidation issues. Even here I'm wondering, if the table changes during the request (could be a CLI process) we'd need a way to clear the cache.